br/pkg/task/restore.go

```diff
@@ -822,38 +835,92 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		restoreError = runSnapshotRestore(c, mgr, g, cmdName, &snapshotRestoreConfig)
 	}
 	if restoreError != nil {
+		// if err happens at register phase no restoreID will be generated and default is 0.
+		if cfg.RestoreID == 0 {
+			return errors.Trace(restoreError)
+		}
+		// if checkpoint is not persisted, let's just unregister the task since we don't need it
+		if hasCheckpointPersisted(c, cfg) {
+			log.Info("pausing restore task from registry",
+				zap.Uint64("restoreId", cfg.RestoreID), zap.Error(restoreError))
+			if err := restoreRegistry.PauseTask(c, cfg.RestoreID); err != nil {
```

Comment:
What if BR is killed because OOM? BR might not have chance to pause the task.

Reply:
good call, let me think about that